src/frontend/apps/deposit/components/Dashboard/DashboardInstructor/index.tsx

```diff
@@ -0,0 +1,187 @@
+import {
+  useDepositedFiles,
+  useUpdateFileDepository,
+} from 'apps/deposit/data/queries';
+import { FileDepository } from 'apps/deposit/types/models';
+
+const PAGE_SIZE = 10;
+
+const messages = {
+  fetchFilesError: {
+    defaultMessage: 'Error fetching files',
+    description: 'Error message when fetching files.',
+    id: 'apps.deposit.components.DashboardInstructor.fetchFilesError',
+  },
+  readFilterOptionsAll: {
+    defaultMessage: 'All',
+    description: 'Filter option for all files.',
+    id: 'apps.deposit.components.DashboardInstructor.readFilterOptionsAll',
+  },
+  readFilterOptionsUnread: {
+    defaultMessage: 'Unread',
+    description: 'Filter option for unread files.',
+    id: 'apps.deposit.components.DashboardInstructor.readFilterOptionsUnread',
+  },
+  readFilterOptionsRead: {
+    defaultMessage: 'Read',
+    description: 'Filter option for read files.',
+    id: 'apps.deposit.components.DashboardInstructor.readFilterOptionsRead',
+  },
+  readFilterTitle: {
+    defaultMessage: 'Filter files',
+    description: 'Title for filter files select.',
+    id: 'apps.deposit.components.DashboardInstructor.readFilterTitle',
+  },
+  readFilterPlaceholder: {
+    defaultMessage: 'Filter',
+    description: 'Placeholder for filter files select.',
+    id: 'apps.deposit.components.DashboardInstructor.readFilterPlaceholder',
+  },
+  filesListHeader: {
+    defaultMessage: 'Students files',
+    description: 'Header for student files list.',
+    id: 'apps.deposit.components.DashboardInstructor.filesListHeader',
+  },
+};
+
+interface DashboardInstructorProps {
+  fileDepository: FileDepository;
+}
+
+export const DashboardInstructor = ({
+  fileDepository,
+}: DashboardInstructorProps) => {
+  const intl = useIntl();
+  const [depositedFilesOffset, setDepositedFilesOffset] = useState(0);
+  const [indices, setIndices] = useState([0, PAGE_SIZE]);
+  const [readFilter, setReadFilter] = useState<Maybe<string>>(undefined);
+
+  const { data, isError, isLoading, refetch } = useDepositedFiles(
+    fileDepository.id,
+    { limit: PAGE_SIZE, offset: depositedFilesOffset, read: readFilter },
+    {
+      refetchInterval: false,
+      refetchIntervalInBackground: false,
+      refetchOnWindowFocus: false,
+      keepPreviousData: true,
+    },
+  );
+  const readFilterOptions = [
+    {
+      label: intl.formatMessage(messages.readFilterOptionsAll),
+      value: undefined,
+    },
+    {
+      label: intl.formatMessage(messages.readFilterOptionsUnread),
+      value: 'false',
+    },
+    {
+      label: intl.formatMessage(messages.readFilterOptionsRead),
+      value: 'true',
+    },
+  ];
+
+  const onReadFilterChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setReadFilter(event.target.value);
+    refetch();
+  };
+
+  const onPageChange = async (event: any) => {
+    setDepositedFilesOffset(event.startIndex);
+    setIndices([event.startIndex, Math.min(event.endIndex, data!.count)]);
```

Comment:
no need to `refetch` here ?

Reply:
No, when setting depositedFilesOffset state, the fetch is automatically done again.